packages/resource-deployment/scripts/run-command-on-all-vmss-for-pool.sh

```diff
@@ -76,67 +80,54 @@ getVmssInfo() {
         . "${0%/*}/wait-for-deployment.sh" -n "$vmssResourceGroup" -t "1800" -q "az group exists --name $vmssResourceGroup"
 
         vmssQueryConditions="?tags.PoolName=='$pool' && tags.BatchAccountName=='$batchAccountName' && resourceGroup=='$vmssResourceGroup'"
         vmssDeployedQuery="[$vmssQueryConditions && provisioningState!='Creating' && provisioningState!='Updating'].name"
-        . "${0%/*}/wait-for-deployment.sh" -n "$vmssResourceGroup" -t "1800" -q "az vmss list --query \"$vmssDeployedQuery\" -o tsv"
+        vmssCreatedCommand="az vmss list --query \"$vmssDeployedQuery\" -o tsv"
+        
+        . "${0%/*}/wait-for-deployment.sh" -n "$vmssResourceGroup" -t "1800" -q "$vmssCreatedCommand"
 
         vmssName=$(az vmss list --query "[$vmssQueryConditions].name" -o tsv)
+        vmssLocation=$(az vmss list --query "[$vmssQueryConditions].location" -o tsv)
+       
+        echo "Checking vmss status - $vmssName"
         vmssStatus=$(az vmss list --query "[$vmssQueryConditions].provisioningState" -o tsv)
         if [ "$vmssStatus" != "Succeeded" ]; then
             echo "Deployment of vmss $vmssName failed with status $vmssStatus"
             exit 1
         fi
 
-         assignSystemIdentity "$vmssResourceGroup" "$vmssName"
-         az resource wait -n "$vmssName" -g "$vmssResourceGroup" --resource-type "Microsoft.Compute/virtualMachineScaleSets" --exists --updated --timeout "1800" --interval "5"
-
-        addResourceGroupNameTagToVMSS $vmssResourceGroup $vmssName
+        echo "Invoking command $commandName"
+        eval "$command"
 
     done
 }
-
-assignSystemIdentity() {
-    local vmssResourceGroup=$1
-    local vmssName=$2
-
-    systemAssignedIdentity=$(az vmss identity assign --name "$vmssName" --resource-group "$vmssResourceGroup" --query systemAssignedIdentity -o tsv)
-    systemAssignedIdentities+=("$systemAssignedIdentity")
-
-    echo \
-        "VMSS Resource configuration:
-  Pool: $pool
-  VMSS resource group: $vmssResourceGroup
-  VMSS name: $vmssName
-  System-assigned identity: $systemAssignedIdentity
-  "
-}
-
-
-addResourceGroupNameTagToVMSS(){
-    vmssResourceGroup=$1
-    vmssName=$2
-    
-    az resource update --set tags.ResourceGroupName="$resourceGroupName" -g "$vmssResourceGroup" -n "$vmssName" --resource-type "Microsoft.Compute/virtualMachineScaleSets"
-
-    echo "Tag ResourceGroupName=$resourceGroupName was added to $vmssName vmss under $vmssResourceGroup resource group"
-}
-
+. "${0%/*}/set-resource-names.sh"
+
+echo "inside pool setup"
+echo "
+resource group:$resourceGroupName
+batch:$batchAccountName
+pool:$pool
+keyVault: $keyVault
+commandName: $commandName
+"
 # Read script arguments
-while getopts ":r:a:p:" option; do
+while getopts ":r:p:c:n:" option; do
     case $option in
     r) resourceGroupName=${OPTARG} ;;
-    a) batchAccountName=${OPTARG} ;;
     p) pool=${OPTARG} ;;
+    c) command=${OPTARG} ;;
+    n) commandName=${OPTARG} ;;
```

Comment:
probably this is redundant option, if we print command in echo above it may be more informative otherwise

Reply:
command can have sensitive information. Hence, added command name